serverlessworkflow/modules/ROOT/pages/core/timeouts-support.adoc

```diff
@@ -4,49 +4,54 @@
-
-For example, in order to cancel the workflow after an hour of execution, you might add the following snippet. 
+For example, to cancel the workflow after an hour of execution, you must use the following configuration.
 
 .Example of workflow timeout
 [source,json]
-[source,json]
 ----
-"timeouts": {
+{
+  "id": "workflow_timeout",
+  "version": "1.0",
+  "name": "Workflow timeout example",
+  "description": "An example of how workflow timeout works",
+  "start": "printWaitMessage",
+  "timeouts": {
     "workflowExecTimeout": "PT1H"
+  } ...
 }
 ----
 
 == Event timeout
 
-When you define a state in a serverless workflow, you can use the `timeouts` property to configure the maximum time to complete this state.
+When you define a state in a workflow, you can use the `timeouts` property to configure the maximum time to complete this state.
 When that time is overdue, the state is considered timed-out, and the engine continues the execution from this state. The execution flow depends on the state type, for instance,
-a transition to a next state.
-All the properties you can use to configure state timeouts are described in the link:{spec_doc_url}#event-timeout-definition[Serverless Workflow specification].
+a transition to a next state might be executed.
+All the properties that you can use to configure state timeouts are described in the link:{spec_doc_url}#states-timeout-definition[Serverless Workflow specification].
 
 Event-based states can use the sub-property `eventTimeout` to configure the maximum time to wait for an event to arrive.
+This is the only property that is supported in current implementation.
 
-
+[#callback-state-timeout]
 === Callback state timeout
-Callback state can be used when you need to execute an action, in general to call an external service, and wait for an asynchronous response in form of an event, the callback.
+The Callback state can be used when you need to execute an action, in general to call an external service, and wait for an asynchronous response in form of an event, the callback.
 
 Once the response event is consumed, the workflow continues the execution, in general moving to the next state defined in the `transition` property. See more on xref:core/working-with-callbacks.adoc[Callback state in {product_name}].
 
-Since the callback state halts the execution util the event is consumed, you can define an `eventTimeout` for it, and in case the event does not arrive in the defined duration time, the workflow continues the execution moving to the next state defined in the transition, see the <<callback-state, example>>.
+Since the Callback state halts the execution util the event is consumed, you can configure an `eventTimeout` for it, and in case the event does not arrive in the configured time duration, the workflow continues the execution moving to the next state defined in the transition, see the <<callback-state, example>>.
```

Comment:
```suggestion
Since the Callback state halts the execution until the event is consumed, you can configure an `eventTimeout` for it, and in case the event does not arrive in the configured time duration, the workflow continues the execution moving to the next state defined in the transition, see the <<callback-state, example>>.
```